### DISCUSSION TITLE
Making default behaviour to crawl from open sources only

### DISCUSSION THREAD

**Discussion by muhammadismailkhan0009** [projectdiscovery/subfinder#744] [Ideas]
currently, the default behaviour of subfinder is to find sub-domains from all available resource whether they are open source or require apikey.
My suggestion is: won't it be good idea to add only those open source resources that can directly be queried without any additional parameters e.g. api key.
For example, the default behaviour shall query from these sources only: alienvault, anubis, commoncrawl,crtsh, dnsdumpster, hackertarget, rapiddns, riddler, and sitedossier.

**Comment by olearycrew** [comment]
I'm not sure I follow. If you add API keys, then subfinder will use those sources...otherwise it uses the default sources that we are confident in.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
